src/resources/parser/glb-parser.js

```diff
@@ -182,22 +185,81 @@ var generateNormals = function (sourceDesc, vertexDesc, positions, numVertices,
     var normals = new Float32Array(normalsTemp.length);
     normals.set(normalsTemp);
 
-    vertexDesc.push({
-        semantic: SEMANTIC_NORMAL,
-        components: 3,
-        type: TYPE_FLOAT32
-    });
-
     sourceDesc[SEMANTIC_NORMAL] = {
         buffer: normals.buffer,
         size: 12,
         offset: 0,
         stride: 12,
-        count: numVertices
+        count: numVertices,
+        components: 3,
+        type: TYPE_FLOAT32
+    };
+};
+
+var flipTexCoordVs = function (vertexBuffer) {
```

Comment:
t would be probably simpler and faster to pass semantic as a parameter to this function, and call it twice (uv0, uv1) then doing array in array loop when copying

Reply:
Not sure I follow. This function could be called twice, but as far as I can see that just moves code from here to the caller. Perhaps I misunderstand?

Reply:
I take it back. I was looking at this

```
var flip = function (offsets, type, one) {
        var typedArray = new type(vertexBuffer.storage);
        for (i = 0; i < offsets.length; ++i) {
            var index = offsets[i].offset;
            var stride = offsets[i].stride;
            for (j = 0; j < vertexBuffer.numVertices; ++j) {
                typedArray[index] = one - typedArray[index];
                index += stride;
            }
        }
    };
```
and thought inside look is the one that loops over offsets. It's the other way, so all good.

Reply:
kk